searx/preferences.py

```diff
@@ -199,6 +212,26 @@ def save(self, name: str, resp: flask.Response):
             resp.set_cookie(name, self.key, max_age=COOKIE_MAX_AGE)
 
 
+class BooleanSetting(Setting):
+    """Setting of a boolean value that has to be translated in order to be storable"""
+
+    def get_bool_value(self, data):
+        for v_str, v_obj in MAP_STR2BOOL.items():
+            if data == v_str:
+                return v_obj
+        raise ValueError("Invalid value for a boolean setting!")
```

Comment:
the intention of the OrderedDict that maps str to boolean is to normalize the str representation of the boolean value .. this function does the opposite and can be replaced `MAP_STR2BOOL[data]` .. I think this issue was unintentional :) .. I will fix it in a WIP commit.

When we store `self.key = data` (see line 227) we store the value from the HTML form in the cookie ..

![image](https://github.com/searxng/searxng/assets/554536/2d104984-e8e1-4421-b069-df4640a97c11)

what we want to store is the first representation of boolean `True` in the OrderedDict, what is 1:

![image](https://github.com/searxng/searxng/assets/554536/788e8c17-b16c-4ece-aece-d64b5708493c)

Reply:
Makes sense, thanks for pointing that out!